VoiceController.php

```diff
@@ -1,48 +1,53 @@
-public function voice(Request $request){
-    $request->validate([
-        'question_id'=>'required|int|exists:questions,id',
-        'value'=>'required|boolean',
-    ]);
+<?php
+public function voice(VoiceRequest $request){
+    
+    // Moved requset to validation
+    $request->validated();
 
+    // FindOrFail is suitable here? Never used in api
     $question=Question::find($request->post('question_id'));
-    if (!$question)
-        return response()->json([
-            'status'=>404,
-            'message'=>'not found question ..'
-        ]);
-    if ($question->user_id==auth()->id())
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote to your question'
-        ]);
+
+        if (!$question)
+            return response()->json([
+                'status'=>404,
+                'message'=>'not found question ..'
+            ]);
+
+        if ($question->user_id==auth()->id())
+            return response()->json([
+                'status' => 500,
+                'message' => 'The user is not allowed to vote to your question'
+            ]);
 
     //check if user voted 
     $voice=Voice::where([
         ['user_id','=',auth()->id()],
         ['question_id','=',$request->post('question_id')]
-    ])->first();
-    if (!is_null($voice)&&$voice->value===$request->post('value')) {
+    ]);
+
+    if ($voice->exists()) {
```

Comment:
@debjit not sure I agree with this change, so you do a double action - `->exists()` and then `->first()`

Reply:
Yes, it is a double call. And I think instead of ->first() we can just directly do $voice->update([]) to it. It will eliminate 2 lines of code.